File: src/main/java/com/EduTrack/EDUTRACK_Portfolio_MS/model/LearningUnitAssessment.java
package com.EduTrack.EDUTRACK_Portfolio_MS.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class LearningUnitAssessment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "assessment_id")
    private Assessment assessment;

    @Enumerated(EnumType.STRING)
    private UnitAssessmentStatus status;

    public enum UnitAssessmentStatus {
        PENDING, COMPLETED, IN_PROGRESS
    }
}

File: frontend/src/app/core/models/learning-unit-assessment.ts
import { Assessment } from "./assessment";

export interface LearningUnitAssessment {
    id?: number;
    assessment: Assessment;
    status: UnitAssessmentStatus;
    canEqual?: boolean;
    hashCode?: boolean;
    toString?: boolean;
}

export interface LearningUnitAssessmentBuilder {
    id: number;
    assessment: Assessment;
    status: UnitAssessmentStatus;
}

export enum UnitAssessmentStatus {
    PENDING = 'PENDING',
    IN_PROGRESS = 'IN_PROGRESS',
    COMPLETED = 'COMPLETED',
    CANCELLED = 'CANCELLED'
}